### PR TITLE
Fix `this.error` with `pos` in `transform` hook

### LINKED ISSUE #5044: "Cannot destructure property 'line' of 'locate(...)' as it is undefined" happens when `pos` is passed to `this.error`
### Rollup Version

3.25.1

### Operating System (or Browser)

stackblitz

### Node Version (if applicable)

_No response_

### Link To Reproduction

https://stackblitz.com/edit/rollup-repro-1umn8w?file=rollup.config.js

### Expected Behaviour

The correct error is output.

### Actual Behaviour

```
[!] (plugin vite:import-analysis-build) TypeError: Cannot destructure property 'line' of 'locate(...)' as it is undefined.
```
happens

refs https://github.com/vitejs/vite/issues/13571

### PR DESCRIPTION
**Pull request by sapphi-red**
This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

resolves #5044
resolves https://github.com/vitejs/vite/issues/13571

### Description

The original source was passed to `augmentCodeLocation` instead of the current source.

### CODE REVIEW

File: test/function/samples/plugin-error-transform-pos/_config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		]
+	},
+	error: {
+		code: 'PLUGIN_ERROR',
+		message: 'error',
+		plugin: 'plugin2',
+		hook: 'transform',
+		id: path.join(__dirname, 'main.js'),
+		watchFiles: [path.join(__dirname, 'main.js')],
+		frame: "11: export const foo = 'foo'bar\n                             ^",
```
Comment (sapphi-red) [inline test/function/samples/plugin-error-transform-pos/_config.js:28]: I'm not sure why the test fails when `frame` is `1: export const foo = 'foo'bar\n                             ^`.

Comment (TrickyPi) [reply test/function/samples/plugin-error-transform-pos/_config.js:28]: The reason is that `deindent` function is slicing the string from index 1 (see https://github.com/rollup/rollup/blob/833314c1621420e031340c1894e7863e062f7761/test/utils.js#L104). You can use "Template literals" like this
```js
frame: `
1: export const foo = 'foo'bar
                                               ^
`,
```

**Review by lukastaegert — APPROVED** [review]
Thanks a lot for spotting this! I took the liberty of adjusting the test as suggested by @TrickyPi, looks good to me otherwise.

File: test/function/samples/plugin-error-transform-pos/_config.js (hunk 1 above)
Comment (sapphi-red) [reply test/function/samples/plugin-error-transform-pos/_config.js]: Ah, I see. Thank you!

**Pull request merged**